core/tests/puppeteer/lighthouse_setup.js

```diff
@@ -160,6 +175,58 @@ const getExplorationEditorUrl = async function(browser, page) {
     await page.click(createButtonSelector);
     await page.waitForSelector(
       dismissWelcomeModalSelector, {visible: true});
+
+    await page.click(dismissWelcomeModalSelector);
+    await page.waitForSelector(stateEditSelector);
+    await page.click(stateEditSelector);
+
+    await page.waitForSelector(stateEditorInput);
+    await page.waitForSelector(saveContentButton);
+    await page.click(saveContentButton);
+    await page.waitForSelector(addInteractionButton);
+    await page.click(addInteractionButton);
+
+    await page.waitForSelector(endIneractionSelector);
+    await page.click(endIneractionSelector);
+    await page.waitForSelector(saveInteractionButton);
+    await page.click(saveInteractionButton);
+
+    await page.waitForSelector(saveChangesButton);
+    await page.click(saveChangesButton);
+
+    await page.waitForSelector(saveDraftButton);
+    await page.click(saveDraftButton);
+
+    const successMessage = 'Changes saved.';
+    let statusMessage;
+    do {
+      await new Promise(r => setTimeout(r, 1000));
+      statusMessage = await page.evaluate(() => {
+        const statusMessageElement = document
+          .querySelector('.e2e-test-toast-message');
+        return statusMessageElement ? statusMessageElement
+          .textContent.trim() : '';
+      });
+    } while (statusMessage !== successMessage);
```

Comment:
I see this pattern being followed in other places in this file, so I'm fine with it! Could it lead to an infinite loop? Do we need do something to avoid it?

Reply:
@DubeySandeep yes we are using this, please see line 383 to generate data for classroom.